src/lib/io/ABC.cpp

```diff
@@ -0,0 +1,429 @@
+                    for (std::size_t i=0; i < dataVec.size(); ++i)
+                    {
+                        indexMap[i] = iData->registerIndexedStr(it->second, dataVec[i].c_str());
+                    }
+
+                    // now remap our indices according to what was registered
+                    for (std::size_t i=0; i < numPts; ++i)
+                    {
+                        int * ptr = iData->dataWrite<int>(it->second, i);
+
+                        std::size_t idx = i;
+
+                        // more points than we have data, so copy over the last
+                        // extent worth of data
+                        if (i >= indexVec.size())
+                        {
+                            idx = indexVec.size() - 1;
+                        }
+                        *ptr = indexMap[indexVec[idx]];
+                    }
+                }
+            }
+            else
+            {
+                std::vector<float> dataVec(dataPts * extent);
+                dataProp.getAs(dataVec.data(), Alembic::Util::kFloat32POD, sel);
+
+                // do we need to multiply the matrix by our points?
+                if (it->first == "P" &&
+                    oMatStack[oMatStack.size() - 1] != Imath::M44d())
+                {
+                    Imath::M44d mat = oMatStack[oMatStack.size() - 1];
+                    for (std::size_t i = 0; i < dataPts; ++i)
+                    {
+                        Imath::V3f v(dataVec[i*3], dataVec[i*3+1], dataVec[i*3+2]);
+                        v = v * mat;
+                        dataVec[i*3] = v.x;
+                        dataVec[i*3+1] = v.y;
+                        dataVec[i*3+2] = v.z;
+                    }
+                }
+
+                for (std::size_t i=0; i < numPts; ++i)
+                {
+                    float * ptr = iData->dataWrite<float>(it->second, i);
+
+                    std::size_t idx = i * extent;
+                    // more points than we have data, so copy over the last
+                    // extent worth of data
+                    if (i >= dataPts)
+                    {
+                        idx = dataVec.size() - extent;
+                    }
+
+                    memcpy(ptr, &(dataVec[idx]), extent * sizeof(float));
+                }
+            }
+        }
+    }
+
+    for (std::size_t i = 0; i < iObj.getNumChildren(); ++i)
+    {
+        Abc::IObject child = iObj.getChild(i);
+        writePoints(child, iTime, iAttrMap, iData, oIndex, oMatStack);
+    }
+
+    if (shouldPop)
+    {
+        oMatStack.pop_back();
+    }
+}
+
+ParticlesDataMutable* readABC(const char* filename,
+                              const bool headersOnly,
+                              std::ostream* errorStream)
+{
+    // partio looks for the extension .abc to decide to call readABC
+    // and we want to be able to accept a time to read (right now in seconds)
+    // so we will pass in /tmp/foo@42.5.abc
```

Comment:
Oh, bummer. That's really unfortunate. Would it help at all if we let you pass in something that looked like a URL? eg. `foo.abc?time=42.5`. Just thinking out loud for the future..

If we start to support this construct now then we will have to forever support it, so we should think about it.